addons/mail/models/mail_message.py

```diff
@@ -1090,3 +1093,18 @@ def _invalidate_documents(self, model=None, res_id=None):
 
     def _get_search_domain_share(self):
         return ['&', '&', ('is_internal', '=', False), ('subtype_id', '!=', False), ('subtype_id.internal', '=', False)]
+
+    def _is_empty_message(self):
+        self.ensure_one()
+        is_body_empty = not self.body or is_html_empty(self.body)
+        is_subtype_description_empty = not self.subtype_id or not self.subtype_id.description or is_html_empty(html_sanitize(self.subtype_id.description, strip_style=True, strip_classes=True))
+        return is_body_empty and is_subtype_description_empty and not self.attachment_ids and not self.tracking_value_ids
+
+    def _clean_related_data(self):
```

Comment:
Would move into the "DISCUSS API" section of the file, this is quite an important method